invenio_rdm_records/resources/serializers/schemaorg/schema.py

```diff
@@ -442,6 +459,44 @@ def get_citation(self, obj):
         ids = _serialize_identifiers(relids)
         return ids or missing
 
+    def _is_dataset(self):
+        return self.context.get("type") == "https://schema.org/Dataset"
```

Comment:
major: since marshmallow field serialization order is not deterministic, it might be that `get_type` isn't called before `get_measurement_techniques` (or `get_distribution`), which means that the `"type"` won't be set in the context.

Maybe it makes sense to make `get_type` a cached property or similar. Then it wouldn't matter who accesses it first.

Reply:
Good point, it turns out the expensive call is already being cached, therefore I don't think we need to cache the get_type method, see [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/serializers/utils.py#L23) the call to read_all which is already cached. I just removed the context and use the `get_type` directly, please let me know if you have a better suggestion